### DISCUSSION TITLE
5. Bundling External Files

### DISCUSSION THREAD

**Discussion by olavea** [olavea/gatsby-transformer-cloudinary-testing-testing-1-2-1-2#13] [Q&A]
Question 5.1: Is this an Old Gatsby Way look-a-like? 
Find this question inline in the code below

Question 5.2: Is this an Old Gatsby Way look-a-like? 
Find this question inline in the code below
Find the code below in [Queen @raae 's Pull Request ](https://github.com/raae/gatsby-transformer-cloudinary/blob/8964d474823e692e75f39e098f4e461c5b0b77e3/packages/gatsby-transformer-cloudinary/gatsby-image/index.js)

[Compare with Gatsby-v4 plugin upgrade Docs Step 5.](https://www.gatsbyjs.com/docs/reference/release-notes/migrating-source-plugin-from-v3-to-v4/#5-bundling-external-files) 

```js
// Cap-Ola-Vea: Question 5.1: Is this an Old Gatsby Way look-a-like? 
const fs = require('fs-extra');

const {
  getFixedImageObject,
  getFluidImageObject,
} = require('./get-image-objects');
const { gatsbyImageTypes } = require('./types');

exports.addFragments = async ({ store, getNodesByType }) => {
  const program = store.getState().program;

  // Check if there are any CloudinaryAsset nodes. If so add fragments for CloudinaryAsset.
  // The fragment will cause an error if there are no CloudinaryAsset nodes.
  if (getNodesByType(`CloudinaryAsset`).length == 0) {
    return;
  }

  // We have CloudinaryAsset nodes so let’s add our fragments to .cache/fragments.
  // Cap-Ola-Vea: Question 5.2: Is this an Old Gatsby Way look-a-like? 
  await fs.copy(
    require.resolve(`./fragments.js`),
    `${program.directory}/.cache/fragments/cloudinary-asset-fragments.js`
  );
};


```

**Comment by raae** [comment]
Seems like it, good catch 💪
